executor/executor.go

```diff
@@ -332,23 +335,20 @@ func (executor *Executor) execute(
 		}
 
 		// short pooling to trigger pod status with max 15 minutes
-
 		statusMap := make(map[string]string)
-		statusMap["Running"] = "pod restarted successfully"
-		statusMap["Failed"] = "pod failed to restart"
-		statusMap["Unknown"] = "pod status is unknown"
-		statusMap["PodInitializing"] = "pod restarting"
+		statusMap[string(kv1.PodRunning)] = "pod restarted successfully"
+		statusMap[string(kv1.PodFailed)] = "pod failed to restart"
+		statusMap[string(kv1.PodUnknown)] = "pod status is unknown"
+		statusMap[string(kv1.PodInitialized)] = "pod restarting"
 
-		backoff := 15 * time.Minute
 		msg := "pod restarting exceeded timout (15 min)"
 		start := time.Now()
-		timeout := int(backoff.Seconds())
 
-		for time.Now().Second() - start.Second() < timeout {
+		for time.Now().Sub(start) < decisionsExecutionTimeout {
 
-			status := "Pending"
+			status := string(kv1.PodPending)
```

Comment:
There is no reason turning this status to string then when checking the status later casting the pending phase to string again would be better replacing the string conversion from the image entirely by setting the statusMap to `map[kv1.PodPhase]string`.

@mohamedveron

Reply:
i am using statues which is not from podPhase tconstants but i will change it